### PR TITLE
Fix for build errors when building with gcc 13

### LINKED ISSUE #1288: cpp_client_telemetry fails to build with gcc13
**Describe your environment.** 
Ubuntu Linux 24.04 with gcc 13

**Steps to reproduce.**
./build.sh

There are files which reference types and don't include necessary header files. It is likely that these header files were being inherited through other included headers and now they must be included explicitly in the project.

The first few encountered:
Enums.hpp: `uint64_t` requires `<cstdint>`
EventFilterCollection.cpp: `std::invalid_argument` requires `<stdexcept>`

The fix is to just go through building the repo repeatedly and adding missing include directives.

### PR DESCRIPTION
**Pull request by DSMitten**
A few files use types defined in headers that are missing, and this is revealed through build errors when building with gcc. Something about libstdc++ may have refactored header files such that headers that previously were inherited now must be explicitly included.

fixes #1288

### CODE REVIEW

File: tests/unittests/BondSplicerTests.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@ class ShadowBondSplicer : protected MAT::BondSplicer
     {
         FullDumpBinaryBlob output;
         static_cast<std::vector<uint8_t>&>(output) = MAT::BondSplicer::splice();
-        return std::move(output);
```
Comment (DSMitten) [inline tests/unittests/BondSplicerTests.cpp:35]: The compiler flagged this as an error, saying that it is a redundent call to `std::move`

Comment (mkoscumb) [reply tests/unittests/BondSplicerTests.cpp:35]: Probably pessimizing move. Good fix.

**Review by mkoscumb — APPROVED** [review]
LGTM

**Review by lalitb — APPROVED** [review]

File: tests/unittests/BondSplicerTests.cpp (hunk 1 above)
Comment (ThomsonTan) [reply tests/unittests/BondSplicerTests.cpp:35]: Seems return value optimization does apply here so explicit move is not required.

**Comment by ThomsonTan** [comment]
Seems all iOS build are broken, are they relatied?

**Comment by lalitb** [comment]
> Seems all iOS build are broken, are they relatied?

#1287

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**